care/facility/api/viewsets/patient_sample.py

```diff
@@ -41,23 +43,18 @@ class PatientSampleViewSet(UserAccessMixin, viewsets.ModelViewSet):
     filterset_class = PatientSampleFilterSet
     http_method_names = ["get", "post", "patch", "delete"]
 
-    def get_serializer(self, *args, **kwargs):
-        """
-        Injects the patient data to serializer
-        """
-        try:
-            kwargs["data"]["patient"] = self.kwargs["patient_pk"]
-        except KeyError:
-            # In read serializers, "data" will not be present.
-            pass
-        return super().get_serializer(*args, **kwargs)
-
     def get_serializer_class(self):
         serializer_class = self.serializer_class
         if self.action == "retrieve":
             serializer_class = PatientSampleDetailSerializer
         return serializer_class
 
+    def get_queryset(self):
+        queryset = super(PatientSampleViewSet, self).get_queryset()
+        if self.kwargs.get("patient_pk") is not None:
```

Comment:
Any reason this is not
`if not self.kwargs.get("patient_pk"):` ?

Reply:
Readability is more in this code. Both are executed same only.

Reply:
In my eyes it's redundant, unless we expect different set of values equating to `if not`, as in it could be either `None` or an empty list and we need to act differently on either cases. Personal preference I guess :)